### PR TITLE
Added Docker troubleshooting tips and code highlighting

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Added Docker troubleshooting tips and code highlighting to make things a bit easier to read.

**Note**: the code highlighting change will fix all the `sh`, `shell`, and `bash` code samples as well, which is a nice bonus 🎉 

|  Before | After  | 
|---|---|
|  Screenshot 2023-12-29 at 11 01 15 AM | Screenshot 2023-12-29 at 11 01 25 AM  | 
|  Screenshot 2023-12-29 at 11 01 48 AM | Screenshot 2023-12-29 at 11 02 07 AM  | 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/deployment/docker.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ package with `yarn build:backend`.
 
 In a CI workflow it might look something like this, from the root:
 
-```bash
+```sh
```
Comment (freben) [inline docs/deployment/docker.md:42]: Did this need to be changed now that bash is among the additional languages?

**Review by freben — APPROVED** [review]

File: docs/deployment/docker.md (hunk 1 above)
Comment (awanlin) [reply docs/deployment/docker.md:42]: Honestly, changed it before as everything else was `sh`. Happy to change it back

Comment (awanlin) [reply docs/deployment/docker.md]: Fixed 👍

**Comment by awanlin** [comment]
Thanks @freben! I've merged this seeing as it's a nice improvement, no point waiting 👍

**Pull request merged**